cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -122,6 +125,42 @@ public void start() {
       Threads.createDaemonThreadFactory("polling-booking-executor"));
 
     store = storeFactory.create();
+
+    initializeScheduleStore();
+  }
+
+  /**
+   * Initialize the stream size schedule store, and restart all the schedule tasks that were persisted in it.
+   *
+   * @throws SchedulerException if the persistent store could not be initialized, or if we couldn't list the
+   *                            persisted tasks
+   */
+  private void initializeScheduleStore() throws SchedulerException {
+    try {
+      scheduleStore.initialize();
+    } catch (Throwable t) {
+      throw new SchedulerException("Error when initializing store for stream size schedules", t);
+    }
+
+    // Reschedule the persisted schedules
+    List<DatasetBasedStreamSizeScheduleStore.StreamSizeScheduleState> scheduleStates;
```

Comment:
why is StreamSizeScheduleState an inner class of the store? Isn't it more closely tied to a ScheduleTask?